src/agreement/aggr_agreement.rs

```diff
@@ -0,0 +1,100 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+use crate::aggregator::AggrSignature;
+use crate::commons::{RoundUpdate, Topics};
+use crate::messages::{payload, Header, Message, Payload};
+use crate::user::committee::CommitteeSet;
+use crate::user::sortition;
+use crate::util::cluster::Cluster;
+use std::sync::Arc;
+use tokio::sync::Mutex;
+use tracing::debug;
+
+use super::{accumulator, verifiers};
+
+pub async fn verify(
+    ru: &RoundUpdate,
+    committees_set: Arc<Mutex<CommitteeSet>>,
+    msg: &Message,
+) -> Result<(), super::verifiers::Error> {
+    if let Payload::AggrAgreement(p) = &msg.payload {
+        let hdr = &msg.header;
+
+        debug!("collected aggr agreement");
+
+        verifiers::verify_votes(
+            hdr.block_hash,
+            p.bitset,
+            p.aggr_signature,
+            committees_set.clone(),
+            sortition::Config::new(ru.seed, ru.round, hdr.step, 64),
+        )
+        .await?;
+
+        // Verify agreement TODO:: new_agreement
+        let m = Message {
+            header: msg.header,
+            payload: Payload::Agreement(p.agreement.clone()),
+            metadata: Default::default(),
+        };
+
+        verifiers::verify_agreement(m, committees_set.clone(), ru.seed).await?;
+        tracing::debug!("valid aggr agreement");
+
+        return Ok(());
+    }
+
+    Err(verifiers::Error::VerificationFailed)
+}
+
+/// Aggregates a list of agreement messages and creates a Message with AggrAgreement payload.
+pub async fn aggregate(
+    ru: &RoundUpdate,
+    committees_set: Arc<Mutex<CommitteeSet>>,
+    agreements: &accumulator::Output,
+) -> Message {
+    let first_agreement = agreements.iter().next().expect("empty agreements");
```

Comment:
Avoid to use `expect` and prefer a Result approach

Furthermore, if you really need to use `expect` you should change the message in something similar to "agreements to not be empty", so that can be read as "Expect agreements to not be empty"

Reply:
This is valid for the whole PR

Reply:
I chose here `except` over `Result` as this should fail-fast.